### PR TITLE
When testing on the expanded matrix, run tests on M1 Mac (via custom runner) as well.

### PR DESCRIPTION
**Pull request by jonsimantov**
If running tests against the expanded matrix, it will build an ARM64 version of the integration tests, and run it on an M1 Mac GitHub runner that we host ourselves.

For security purposes, this job does not use any third-party marketplace actions, nor does it pass in secrets on the command line.

The test_desktop_custom_runner step is written as generically as possible to support other custom runners we may add in the future.
***

Run integration-tests workflow on the expanded matrix. This works both on prebuilt SDK and from source.
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [X] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

**Comment by jonsimantov** [comment]
@sunmou99 I'm still considering this a draft change because I want to make sure that the runner is stable before merging this, but could you take a look at the changes I've made in this PR and give me your comments?

In particular I am wondering if it would be better to separate out architecture in the summarize script rather than the hack I've done. I'm intending to add x86 integration tests soon, so maybe that would be a better time to make that change.

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -203,7 +221,7 @@ jobs:
           --run_id ${{github.run_id}}
 
   build_desktop:
-    name: build-desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}
+    name: build-desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}-${{ matrix.arch }}
```
Comment (sunmou99) [inline .github/workflows/integration_tests.yml:224]: How about reverse the order of the last 2 variables?
build-desktop-${{ matrix.os }}-${{ matrix.arch }}-${{ matrix.ssl_variant }}

File: scripts/gha/summarize_test_results.py [hunk 2]
```diff
@@ -75,6 +75,7 @@
 
 CAPITALIZATIONS = {
     "macos": "MacOS",
+    "macosarm": "MacOS-ARM",
```
Comment (sunmou99) [inline scripts/gha/summarize_test_results.py:78]: I prefer to have only one MacOS, and make arch as a secondary description. Some more extra work is required here for the result summary report. I'll figure it out.

Comment (jonsimantov) [reply scripts/gha/summarize_test_results.py:78]: Do you want to work in this same branch?

File: .github/workflows/integration_tests.yml (hunk 1 above)
Comment (jonsimantov) [reply .github/workflows/integration_tests.yml]: Done.

**Pull request closed without merging**